docs/design/uptime-reputation-calculation.md

```diff
@@ -0,0 +1,122 @@
+
+This design doc is focused on how to design and implement a fair, simple and understandable uptime reputation calculation after our initial approach didn't behave as expected, causing the disqualification of several nodes without having a proof that they have failed enough uptime checks that the _SNs_ were offline long time enough which put in danger the minimum level of the network availability and retrievability.
+
+## Design
+
+The new process for doing the uptime checks is based on once an _SN_ is selected for having the check, if the _SN_ fails the initial check, then it's tracked to periodically receive uptime checks until (a) it successfully passes an uptime check or (b) gets disqualified due to many uptime check failures in one month.
+
+Once an _SN_ has failed the first uptime check, the _SN_ is considered offline and the subtotal time, which the _SN_ is considered offline, is the time elapsed between the first and the last failed uptime check. The total offline time of an _SN_ is the accumulation of the subtotals that the _SN_ has had during the ongoing month and when such total surpasses the maximum permitted percentage time, the _SN_ will get disqualified.
+
+The maximum percentage of time that an _SN_ can be offline will be established with the network commitment in terms of availability and retrievability during one month.
+
+The process is divided in 2 parts:
+
+1. Selecting _SNs_ for receiving uptime checks.
+1. Periodically check the _SNs_ which has failed the first uptime check until they pass one or get disqualified.
+
+For the first part, we foresee to use the current implemented mechanism with some minor modifications for excluding the _SNs_ which are being checked by the second part.
+
+The second part requires a totally new implementation which should be totally independent of any current satellite service for being able in the near future to run it separately for scalability purpose.
+
+
+## Rationale
+
+The approach described in this document present the following advantages:
+
+1. It's clear why an _SN_ get disqualified due to failed uptime checks.
+1. It has a clear proof for an _SNOs_ why their _SNs_ get disqualified due to failed uptime check in case that they request it before the designed time to clean up the historical failed uptime checks data.
+1. It allows to report to _SN_ at any how far is to be disqualified by failing uptime checks.
+
+And the following disadvantages
+
+1. It requires to store, for some period of time, historical for nodes which have failed at least one uptime check.
+1. It requires more resources than the current implemented approach.
+1. It doesn't calculate the exact time that an _SN_ is offline; once an _SN_ has a failed uptime check the time elapsed between the last failed uptime check and the next succeeded uptime check causes not to have the accurate offline time.
```

Comment:
If we coordinate with the Kademlia Removal (https://github.com/storj/storj/pull/2686) we can have a plan for the storage node to notify the node when it comes back online.  This should reduce this concern.

Reply:
Yes, I thought about it and I asked for confirmation in https://github.com/storj/storj/pull/2733#discussion_r312189487